### PR TITLE
Zc/iteration 4 backend testing

### PR DESCRIPTION
**Pull request by zclewell**
## Status: 
:rocket: Ready

## Description
Added tests for search routes, modified model.py to prevent flakiness

### CODE REVIEW

File: backend/tests/test_model.py [hunk 1]
```diff
@@ -1,24 +1,22 @@
+import pytest
 import model
-from mongoengine import connect
-from test_data import valid_items, invalid_items
+import test_data
```
Comment (rsun0) [inline backend/tests/test_model.py:3]: I like the import convention switch

File: backend/tests/test_get.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import pytest
```
Comment (rsun0) [inline backend/tests/test_get.py]: Maybe rename this file to test_get_by_keyword? That way, future tests for future GET routes can be put into other files without confusion. I named my test files by their corresponding function names in app.py, but you can shorten the test file name to test_search or something (maybe even rename the app.py function if you want).

File: backend/tests/test_get.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    name = first + second
+
+    rv = client.get('/search?keyword=' + name)
+    response = json.loads(rv.data)
+    first = response[0]
+
+    assert first['upc'] == upc
+
+
+def test_search_by_keyword_partial(client, existing_item):
```
Comment (rsun0) [inline backend/tests/test_get.py]: Great variety of test cases!

File: backend/tests/test_get.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    name = name[:int(len(name) / 2)]
+
+    rv = client.get('/search?keyword=' + name)
+    response = json.loads(rv.data)
+    first = response[0]
+
+    assert first['upc'] == upc
+
+
+def test_search_upc_over_keyword(client):
```
Comment (rsun0) [inline backend/tests/test_get.py]: Good job testing this edge case

**Review by rsun0 — CHANGES_REQUESTED** [review]
Consider renaming test_get.py to something more specific to GET /search.

File: backend/tests/test_get.py (hunk 2 above)
Comment (zclewell) [reply backend/tests/test_get.py]: Completed

**Review by rsun0 — APPROVED** [review]
Thanks for the rename!
